### ISSUE TITLE
vim-x11: terminal background leaks into vim's

### ISSUE THREAD

**Issue by blehel** [void-linux/void-packages#23497]
### System

* xuname:  
  Void 5.4.48_1 x86_64 GenuineIntel notuptodate rrFFFF
* package:  
  vim-x11-8.2.1095_1

### Problem description

After upgrading from 8.2.0891_1 to 8.2.1095_1, in urxvt when opening a file in vim, there are some annoying black spots at the top: 
[example1](https://i.imgur.com/DKbjNnJ.png)
[example2](https://i.imgur.com/esyGF8q.png)
[example3](https://i.imgur.com/8iO0KcR.png)
Previously all of this was the blue color of the background. Pressing CTRL-L to redraw the screen or scrolling down until the affected region goes off-screen does away with the problem.
Note that if I change the terminal background to some other color, that will be the color that creates these spots, so this is about the terminal background not being drawn over.

### Steps to reproduce the behavior

1. Edit your vimrc so that you use a colorscheme with a background different to your terminal's. Mine is a custom one based on `blue`, and my terminal background is black.
So for example add `colorscheme blue` to your vimrc. (setting it from the ex prompt does not bring out the issue).
2. Start `urxvt` (although I've seen similar behaviour - black spots appearing in place of the background color - in the past in terminal emulators which do not support the `bce` /background color erase/ terminal capability, like `kitty`).
3. Open something in vim.

I don't exclude this being an rxvt-unicode problem, but it was the vim upgrade which broke things.

**Comment by NeelChotai** [comment]
I can reproduce this, it seems to be a Vim issue.

I've fixed this by appending `autocmd VimEnter * redraw!` to my vimrc.

**Comment by biopsin** [comment]
@blehel did you try raise this issue over @ https://github.com/vim/vim/issues ?
I quickly search thru their commits, hoping to spot something in this regards,
but they are releasing daily so this might or not be fixed in one of those lately ones..

**Comment by blehel** [comment]
> @blehel did you try raise this issue over @ https://github.com/vim/vim/issues ?
> I quickly search thru their commits, hoping to spot something in this regards,
> but they are releasing daily so this might or not be fixed in one of those lately ones..

No because they were a few releases ahead, and I did not want to report an issue for a release that was not current.
And based on the suggestion above, I found that `autocmd VimEnter * redraw!` provides a workaround for it. 

I checked to see if vim still has the same behavior without this option, and nothing has changed. Did a search through their issues, and found one that basically describes the same problem, even narrowing it down to the exact patch: https://github.com/vim/vim/issues/6607. I'll leave a comment there.
